sourced/ml/mining/cmd/extract_stdlib.py

```diff
@@ -0,0 +1,121 @@
+import argparse
+from io import BytesIO
+import json
+import logging
+from pathlib import Path
+import re
+from zipfile import ZipFile
+import zlib
+
+from lxml import html
+import requests
+from requests.exceptions import HTTPError
+import yaml
+
+from sourced.ml.mining.models import StandardLibraries
+from sourced.ml.mining.utils import check_remove_filepath, path_with_suffix
+
+
+def get_page(url: str):
+    """Return the GET response to the given URL or raise an error."""
+    try:
+        return requests.get(url)
+    except HTTPError as http_err:
+        print("HTTP error occurred: %s" % http_err)
+    except Exception as err:
+        print("Other error occurred: %s " % err)
+
+
+def get_leaves(url: str, xpath: str):
+    """Return the nodes extracted from the GET response's content."""
+    content = get_page(url).content
+    tree = html.fromstring(content)
+    return tree.xpath(xpath)
+
+
+def is_zlib_header(header: bytes):
+    """Decode the input header, return True if successful and False if not."""
+    try:
+        header.decode(encoding="utf-8")
+        return True
+    except UnicodeDecodeError:
+        return False
```

Comment:
These 3 functions make me think that you should rewrite crawling using Scrapy.

Reply:
see below, I don't mind but not sure it's necessary - I abstracted te first two to avoid repetition, and the third because it's used in a loop and was prettier